src/DevBetterWeb.Infrastructure/Services/DailyCheckPingService.cs

```diff
@@ -0,0 +1,150 @@
+
+        var adminPingMessage = await SendAdminPing(invitationsForAdminPing);
+        messages.Append(adminPingMessage);
+      }
+    }
+
+    public List<Invitation> CheckIfAnyActiveInvitationsRequireAdminPing(List<Invitation> invitations)
+    {
+      var invitationsRequiringAdminPing = new List<Invitation>();
+
+      foreach(var invite in invitations)
+      {
+        bool inviteRequiresAdminPing = true;
+
+        if (invite.DateOfUserPing.Equals(DateTime.MinValue)) inviteRequiresAdminPing = false;
+        if (invite.DateOfLastAdminPing.Equals(DateTime.Today)) inviteRequiresAdminPing = false;
+        if (DateTime.Today - invite.DateCreated < new TimeSpan(4, 0, 0, 0)) inviteRequiresAdminPing = false;
+        if (DateTime.Today - invite.DateOfUserPing < new TimeSpan(2, 0, 0, 0)) inviteRequiresAdminPing = false;
+        if (!invite.Active) inviteRequiresAdminPing = false;
+
+        if (inviteRequiresAdminPing) invitationsRequiringAdminPing.Add(invite);
+      }
+
+      return invitationsRequiringAdminPing;
+    }
+
+    public List<Invitation> CheckIfAnyActiveInvitationsRequireUserPing(List<Invitation> invitations)
+    {
+      var invitationsRequiringUserPing = new List<Invitation>();
+
+      foreach(var invite in invitations)
+      {
+        if((DateTime.Today - invite.DateCreated) >= new TimeSpan(2,0,0,0) && invite.DateOfUserPing.Equals(DateTime.MinValue))
+        {
+          invitationsRequiringUserPing.Add(invite);
+        }
+      }
+
+      return invitationsRequiringUserPing;
+    }
+
+    private async Task<List<string>> SendUserPing(List<Invitation> invitations)
+    {
+      var messagesToAdd = new List<string>();
+
+      var emailSubject = "Finish setting up your DevBetter account!";
+      var emailBody = "Don't forget to click the link below to finish setting up your DevBetterAccount!\n";
+
+      foreach (var invitation in invitations)
+      {
+        var url = $"https://devbetter.com/Identity/Account/NewMemberRegister/{invitation.InviteCode}/{invitation.Email}";
+
+        await _emailService.SendEmailAsync(invitation.Email, emailSubject, $"{emailBody}{url}");
+        messagesToAdd.Add($"User at email {invitation.Email} has been reminded to finish setting up their account.");
+        invitation.UpdateUserPingDate();
+      }
+
+      return messagesToAdd;
+    }
+    private async Task<string> SendAdminPing(List<Invitation> invitations)
+    {
+      var emailSubject = "Remind user(s) to finish setting up their DevBetter account(s)";
+      var listOfEmails = "";
+
+      foreach (var invitation in invitations)
+      {
+        listOfEmails += $"{invitation.Email}\n";
+      }
+
+      var emailBody = $"Please remind these users to finish setting up their DevBetter accounts: {listOfEmails}";
+
+      var usersInAdminRole = await _userManager.GetUsersInRoleAsync(AuthConstants.Roles.ADMINISTRATORS);
+
+      var emailList = "";
```

Comment:
What's the difference between `listOfEmails` and `emailList`?

It seems like the latter is the list of sent emails while the former is the list of emails to send. Name them accordingly.